internal/commands/app/create_inputs.go

```diff
@@ -88,29 +85,28 @@ func (i *createInputs) resolveDirectory(wd string) (string, error) {
 		return "", appErr
 	}
 	if appOK {
-		return "", fmt.Errorf("%s is inside or is a Realm app directory", dir)
+		return "", fmt.Errorf("%s is inside or is a Realm app directory", fullPath)
 	}
 	return fullPath, nil
 }
 
-// TODO(REALMC-8135): Implement data-source flag for app create command
-// func (i *createInputs) resolveDataSource(client realm.Client, groupID, appID string) (string, error) {
-// 	clusters, err := client.ListClusters(groupID, appID)
-// 	if err != nil {
-// 		return "", err
-// 	}
-// 	var dsCluster string
-// 	for _, cluster := range clusters {
-// 		if (i.DataSource == "" && cluster.State == "IDLE") || i.DataSource == cluster.Name {
-// 			dsCluster = cluster.Name
-// 			break
-// 		}
-// 	}
-// 	if dsCluster == "" {
-// 		if i.DataSource != "" {
-// 			return "", fmt.Errorf("Unable to find the %s cluster", i.DataSource)
-// 		}
-// 		return "", fmt.Errorf("Unable to find any available cluster for Group ID %s", groupID)
-// 	}
-// 	return dsCluster, nil
-// }
+func (i *createInputs) resolveDataSource(client realm.Client, groupID, appID string) (string, error) {
```

Comment:
do we have tests for this here?  probably somewhat implicitly through the `Handler` tests, but can we write a few unit tests for this to make sure we're returning the errors we expect, and also matching on the `"IDLE"`/`cluster.Name` logic?